### PR TITLE
Refactoring EditorAction

### PR DESCRIPTION
**Pull request by Rafi993**
## What type of PR is this?

- [x] Refactor
- [ ] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

Refactoring editor actions
  1. Removing nested ternary operators for improving readability
  2. Converting class component to functional component

## Related Tickets & Documents
 
closes https://github.com/forem/forem/issues/10947

## QA Instructions, Screenshots, Recordings

1. Create a post and publish it
2. Edit a post
3. Convert a post to draft
4. Create a new post and save it as a draft

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.forem.com
- [ ] readme
- [x] no documentation needed

### CODE REVIEW

**Comment by Rafi993** [comment]
I think the code climate error might be a false positive.

File: app/javascript/article-form/components/EditorActions.jsx [hunk 1]
```diff
@@ -18,102 +19,91 @@ const Icon = () => (
...
-      version,
-      passedData,
-      onConfigChange,
-      submitting,
-    } = this.props;
+  const v1 = version === 'v1';
+  const v2 = version === 'v2';
 
-    const { moreConfigShowing } = this.state;
+  const toggleMoreConfig = useCallback(
```
Comment (nickytonline) [inline app/javascript/article-form/components/EditorActions.jsx:38]: refactor (blocking): `useCallback` seems overkill here. For context, see  [When to useMemo and useCallback](https://kentcdodds.com/blog/usememo-and-usecallback).

File: app/javascript/article-form/components/EditorActions.jsx [hunk 2]
```diff
@@ -18,102 +19,91 @@ const Icon = () => (
...
-      onPublish,
-      onClearChanges,
-      published,
-      edited,
-      version,
-      passedData,
-      onConfigChange,
-      submitting,
-    } = this.props;
+  const v1 = version === 'v1';
```
Comment (nickytonline) [inline app/javascript/article-form/components/EditorActions.jsx:35]: nitpick (non-blocking): consider naming these `isVersion1` and `isVersion2`.

**Review by nickytonline — CHANGES_REQUESTED** [review]
Looking good @Rafi993. A small change request.

**Review by nickytonline — APPROVED** [review]
Tested this locally and it LGTM. 🚀

**Comment by Rafi993** [comment]
> Tested this locally and it LGTM. 🚀

Thank you @nickytonline

**Review by rhymes — APPROVED** [review]

**Pull request merged**